pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -35,6 +39,17 @@ func StatefulSetForCluster(
 	statefulSetName := util.NodePoolResourceName(cluster, np)
 	seedProviderServiceName := util.SeedProviderServiceName(cluster)
 	nodePoolLabels := util.NodePoolLabels(cluster, np.Name)
+
+	datacenter := cassDefaultDatacenter
+	if np.Datacenter != "" {
+		datacenter = np.Datacenter
+	}
```

Comment:
:grey_question: Not suggesting now, but might we also create a (mutating) AdmissionController to do this? Ala https://kubernetes.io/docs/admin/admission-controllers/#defaultstorageclass